closed/src/java.base/share/native/libjncrypto/NativeCrypto.c

```diff
@@ -347,10 +417,21 @@ JNIEXPORT jint JNICALL Java_jdk_crypto_jniprovider_NativeCrypto_loadCrypto
     OSSL_RSA_free = (OSSL_RSA_free_t *)find_crypto_symbol(crypto_library, "RSA_free");
     OSSL_RSA_public_decrypt = (OSSL_RSA_public_decrypt_t *)find_crypto_symbol(crypto_library, "RSA_public_decrypt");
     OSSL_RSA_private_encrypt = (OSSL_RSA_private_encrypt_t *)find_crypto_symbol(crypto_library, "RSA_private_decrypt");
+    OSSL_RSA_get0_key = (OSSL_RSA_get0_key_t *)find_crypto_symbol(crypto_library, "RSA_get0_key");
     OSSL_BN_bin2bn = (OSSL_BN_bin2bn_t *)find_crypto_symbol(crypto_library, "BN_bin2bn");
+    OSSL_BN_mod_exp = (OSSL_BN_mod_exp_t *)find_crypto_symbol(crypto_library, "BN_mod_exp");
+    OSSL_BN_ucmp = (OSSL_BN_ucmp_t *)find_crypto_symbol(crypto_library, "BN_ucmp");
+    OSSL_BN_clear = (OSSL_BN_zero_t *)find_crypto_symbol(crypto_library, "BN_clear");
+    OSSL_BN_CTX_new = (OSSL_BN_CTX_new_t *)find_crypto_symbol(crypto_library, "BN_CTX_new");
+    OSSL_BN_CTX_free = (OSSL_BN_CTX_free_t *)find_crypto_symbol(crypto_library, "BN_CTX_free");
+    OSSL_BN_new = (OSSL_BN_new_t *)find_crypto_symbol(crypto_library, "BN_new");
     OSSL_BN_set_negative = (OSSL_BN_set_negative_t *)find_crypto_symbol(crypto_library, "BN_set_negative");
     OSSL_BN_free = (OSSL_BN_free_t *)find_crypto_symbol(crypto_library, "BN_free");
 
+    if (OSSL_RSA_get0_key == NULL) {
+        OSSL_RSA_get0_key = (OSSL_RSA_get0_key_t *) &OSSL_RSA_get0_key_compatibility;
+    }
```

Comment:
This should not apply for OpenSSL 1.1.*, right? If so, that condition should be included (i.e. `0 != ossl_ver`).

Reply:
Correction: `OSSL_RSA_get0_key_compatibility` should only be used if `0 == ossl_ver`.